automation/Extension/webext-instrumentation/src/lib/js-instruments.ts

```diff
@@ -546,20 +555,23 @@ export function jsInstruments(event_id, sendMessagesToLogger) {
   ) {
     // Store original descriptor in closure
     const propDesc = Object.getPropertyDescriptor(object, propertyName);
-    if (!propDesc) {
-      console.error(
-        "Property descriptor not found for",
-        objectName,
-        propertyName,
-        object,
-      );
-      return;
-    }
+
+    // Property descriptor for undefined properties
+    let undefinedPropValue;
+    const undefinedPropDesc = {
+      get: () => {
+        return undefinedPropValue;
+      },
+      set: value => {
+        undefinedPropValue = value;
```

Comment:
Mind to try adapting the solution in https://github.com/mozilla/openwpm-webext-instrumentation/pull/19/files to see if it prevent prototype object pollution? This might fix the cross-instance changes we see in the `MockClass` test below.